### ISSUE TITLE
tensor: Contiguous has no MSL artifact, so any graph that slices cannot run on Metal

### ISSUE THREAD

**Issue by changkun** [golang-design/accel#19]
Found by running a real model end to end on Metal for the first time, which is the class of thing that only shows up there.

`Contiguous` cannot run on Metal. Its kernel is the only one in the corpus with no MSL artifact:

```
accel: node 589: kernel Pack carries no MSL artifact, so it cannot run on Metal;
it is outside the subset specs/021-metal-bringup.md section 5 lowers
```

Checked across the whole corpus rather than assuming — 66 of 67 kernels carry an `MSL:` field and `Pack` is the exception.

### Why this stops a model rather than costing it something

`Contiguous` is not optional for a transformer. `Slice` returns a view at a non-zero offset, and every elementwise operator and `MatMul` refuses a strided operand — correctly, since silently packing would hide a real copy. So `Contiguous` is the only way to use a slice, and the refusal messages name it.

The forward pass needs it in the one place that has no alternative: slicing the last row of the hidden state before the LM head. Running the head over all T positions instead costs T×V — for Qwen3-4B at T=2000 that is 1.2 GB of logits nobody reads, larger than the model's int8 weights. So the choices on Metal today are a graph that will not compile, or one that allocates more than the model.

The same failure reaches the depthwise convolution I measured for #17 — that composition packs K windows — so it is not confined to one call site.

### What is odd about it

The CPU backend runs `Pack` fine, so the kernel exists and is correct; only the MSL lowering is absent. `specs/025-tensor-operators.md` documents `Contiguous` without qualification, and nothing at graph-build time says the operator is CPU-only. The refusal arrives at **compile**, after the graph is recorded and the weights are on the device — a 1.4 GB upload before the error, in my case.

So there are two asks, and the small one is worth doing even if the large one is not scheduled:

1. **Say it earlier, or say it in the docs.** `Contiguous` on a Metal device could refuse at record time, the way `Attention` refuses an f16 prefill by name. Right now the first signal is a node id after a full load.
2. **The MSL lowering.** Whatever put `Pack` outside `021 §5`'s subset — I would guess the rank-generic index arithmetic over `PackRank` extents and strides in one std140 block, since that is what its doc comment emphasises.

### What I am doing meanwhile

Running on the CPU backend, and reporting Metal as unavailable for the dense path rather than shipping a graph that dies after the upload. Happy to test a fix against the real 596M-parameter model as soon as one exists — the reproducer is one `Contiguous` call in a compiled plan.

**Comment by changkun** [comment]
Fixed on `main`. `Contiguous` runs on Metal, and I took the large ask rather than the small one — the earlier signal you asked for would have been a better refusal for a capability that should not have been refused at all.

**Your diagnosis was right, including the guess.** `PackParams` carries two eight-element arrays, and std140 gives an array member a sixteen-byte stride whatever its element type — so it cannot be one C array the body indexes with one index. The emitter refused array members with the comment *"no corpus kernel needs it yet."* `Pack` did.

It is spelled now the way the *matrix* member beside it already was: an outer array of the element count, each row the stride in elements, with the body appending the inner index — always zero, because the element sits at the start of its slot. The declaration pattern was sitting right there; nothing had connected it to the array case.

```
uint Extent[8][4];
...
uint e = p.Extent[a][0];
```

**Verification, since a pack that runs is not a pack that is right:**

- `Pack` is in the Metal differential now, and the case **transposes a 3×4 matrix** rather than copying one — so a lowering that ignored a stride would produce the same twelve values in the wrong order rather than obviously wrong ones. Bit-for-bit against the CPU oracle.
- Your exact reproducer is a permanent test: slice the last row of a hidden state, `Contiguous` it, compare **exactly** on Metal. A pack copies and computes nothing, so equality is the right assertion.
- Verified load-bearing by dropping the index rewrite — the Metal compiler refuses the program outright.

All fifty-seven corpus kernels now carry an MSL artifact.

**On the thing you noticed that I would not have.** The corpus has a guard that every kernel lowering to MSL must appear in a differential case, and it fired the moment `Pack` started lowering — so the comparison could not be forgotten. But nothing guarded the *other* direction: a kernel with no MSL artifact at all, silently making a public operator CPU-only. That asymmetry is why this reached you instead of CI, and it is worth more than the fix.

**On the ordering complaint, which stands.** You are right that a refusal after a 1.4 GB upload is the wrong place for it. It is moot for `Contiguous` now, and it is not moot in general — any future kernel without MSL would fail the same way. I have not fixed that and am not claiming to.

Please re-test against the 596M model when convenient. If the dense path now runs end to end on Metal, that is worth knowing precisely because you had already reported Metal unavailable — I would rather hear it from your side than infer it from mine.
